oci.entrypoint gets escaped no matter what

### Is there an existing issue for this?

- [x] There is no existing issue for this bug

### Is this happening on an up to date version of Incus?

- [x] This is happening on a supported version of Incus

### Incus system details

```yaml
config:
  core.https_address: :8443
api_extensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_stop_priority
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
- storage
- file_delete
- file_append
- network_dhcp_expiry
- storage_lvm_vg_rename
- storage_lvm_thinpool_rename
- network_vlan
- image_create_aliases
- container_stateless_copy
- container_only_migration
- storage_zfs_clone_copy
- unix_device_rename
- storage_lvm_use_thinpool
- storage_rsync_bwlimit
- network_vxlan_interface
- storage_btrfs_mount_options
- entity_description
- image_force_refresh
- storage_lvm_lv_resizing
- id_map_base
- file_symlinks
- container_push_target
- network_vlan_physical
- storage_images_delete
- container_edit_metadata
- container_snapshot_stateful_migration
- storage_driver_ceph
- storage_ceph_user_name
- resource_limits
- storage_volatile_initial_source
- storage_ceph_force_osd_reuse
- storage_block_filesystem_btrfs
- resources
- kernel_limits
- storage_api_volume_rename
- network_sriov
- console
- restrict_dev_incus
- migration_pre_copy
- infiniband
- dev_incus_events
- proxy
- network_dhcp_gateway
- file_get_symlink
- network_leases
- unix_device_hotplug
- storage_api_local_volume_handling
- operation_description
- clustering
- event_lifecycle
- storage_api_remote_volume_handling
- nvidia_runtime
- container_mount_propagation
- container_backup
- dev_incus_images
- container_local_cross_pool_handling
- proxy_unix
- proxy_udp
- clustering_join
- proxy_tcp_udp_multi_port_handling
- network_state
- proxy_unix_dac_properties
- container_protection_delete
- unix_priv_drop
- pprof_http
- proxy_haproxy_protocol
- network_hwaddr
- proxy_nat
- network_nat_order
- container_full
- backup_compression
- nvidia_runtime_config
- storage_api_volume_snapshots
- storage_unmapped
- projects
- network_vxlan_ttl
- container_incremental_copy
- usb_optional_vendorid
- snapshot_scheduling
- snapshot_schedule_aliases
- container_copy_project
- clustering_server_address
- clustering_image_replication
- container_protection_shift
- snapshot_expiry
- container_backup_override_pool
- snapshot_expiry_creation
- network_leases_location
- resources_cpu_socket
- resources_gpu
- resources_numa
- kernel_features
- id_map_current
- event_location
- storage_api_remote_volume_snapshots
- network_nat_address
- container_nic_routes
- cluster_internal_copy
- seccomp_notify
- lxc_features
- container_nic_ipvlan
- network_vlan_sriov
- storage_cephfs
- container_nic_ipfilter
- resources_v2
- container_exec_user_group_cwd
- container_syscall_intercept
- container_disk_shift
- storage_shifted
- resources_infiniband
- daemon_storage
- instances
- image_types
- resources_disk_sata
- clustering_roles
- images_expiry
- resources_network_firmware
- backup_compression_algorithm
- ceph_data_pool_name
- container_syscall_intercept_mount
- compression_squashfs
- container_raw_mount
- container_nic_routed
- container_syscall_intercept_mount_fuse
- container_disk_ceph
- virtual-machines
- image_profiles
- clustering_architecture
- resources_disk_id
- storage_lvm_stripes
- vm_boot_priority
- unix_hotplug_devices
- api_filtering
- instance_nic_network
- clustering_sizing
- firewall_driver
- projects_limits
- container_syscall_intercept_hugetlbfs
- limits_hugepages
- container_nic_routed_gateway
- projects_restrictions
- custom_volume_snapshot_expiry
- volume_snapshot_scheduling
- trust_ca_certificates
- snapshot_disk_usage
- clustering_edit_roles
- container_nic_routed_host_address
- container_nic_ipvlan_gateway
- resources_usb_pci
- resources_cpu_threads_numa
- resources_cpu_core_die
- api_os
- container_nic_routed_host_table
- container_nic_ipvlan_host_table
- container_nic_ipvlan_mode
- resources_system
- images_push_relay
- network_dns_search
- container_nic_routed_limits
- instance_nic_bridged_vlan
- network_state_bond_bridge
- usedby_consistency
- custom_block_volumes
- clustering_failure_domains
- resources_gpu_mdev
- console_vga_type
- projects_limits_disk
- network_type_macvlan
- network_type_sriov
- container_syscall_intercept_bpf_devices
- network_type_ovn
- projects_networks
- projects_networks_restricted_uplinks
- custom_volume_backup
- backup_override_name
- storage_rsync_compression
- network_type_physical
- network_ovn_external_subnets
- network_ovn_nat
- network_ovn_external_routes_remove
- tpm_device_type
- storage_zfs_clone_copy_rebase
- gpu_mdev
- resources_pci_iommu
- resources_network_usb
- resources_disk_address
- network_physical_ovn_ingress_mode
- network_ovn_dhcp
- network_physical_routes_anycast
- projects_limits_instances
- network_state_vlan
- instance_nic_bridged_port_isolation
- instance_bulk_state_change
- network_gvrp
- instance_pool_move
- gpu_sriov
- pci_device_type
- storage_volume_state
- network_acl
- migration_stateful
- disk_state_quota
- storage_ceph_features
- projects_compression
- projects_images_remote_cache_expiry
- certificate_project
- network_ovn_acl
- projects_images_auto_update
- projects_restricted_cluster_target
- images_default_architecture
- network_ovn_acl_defaults
- gpu_mig
- project_usage
- network_bridge_acl
- warnings
- projects_restricted_backups_and_snapshots
- clustering_join_token
- clustering_description
- server_trusted_proxy
- clustering_update_cert
- storage_api_project
- server_instance_driver_operational
- server_supported_storage_drivers
- event_lifecycle_requestor_address
- resources_gpu_usb
- clustering_evacuation
- network_ovn_nat_address
- network_bgp
- network_forward
- custom_volume_refresh
- network_counters_errors_dropped
- metrics
- image_source_project
- clustering_config
- network_peer
- linux_sysctl
- network_dns
- ovn_nic_acceleration
- certificate_self_renewal
- instance_project_move
- storage_volume_project_move
- cloud_init
- network_dns_nat
- database_leader
- instance_all_projects
- clustering_groups
- ceph_rbd_du
- instance_get_full
- qemu_metrics
- gpu_mig_uuid
- event_project
- clustering_evacuation_live
- instance_allow_inconsistent_copy
- network_state_ovn
- storage_volume_api_filtering
- image_restrictions
- storage_zfs_export
- network_dns_records
- storage_zfs_reserve_space
- network_acl_log
- storage_zfs_blocksize
- metrics_cpu_seconds
- instance_snapshot_never
- certificate_token
- instance_nic_routed_neighbor_probe
- event_hub
- agent_nic_config
- projects_restricted_intercept
- metrics_authentication
- images_target_project
- images_all_projects
- cluster_migration_inconsistent_copy
- cluster_ovn_chassis
- container_syscall_intercept_sched_setscheduler
- storage_lvm_thinpool_metadata_size
- storage_volume_state_total
- instance_file_head
- instances_nic_host_name
- image_copy_profile
- container_syscall_intercept_sysinfo
- clustering_evacuation_mode
- resources_pci_vpd
- qemu_raw_conf
- storage_cephfs_fscache
- network_load_balancer
- vsock_api
- instance_ready_state
- network_bgp_holdtime
- storage_volumes_all_projects
- metrics_memory_oom_total
- storage_buckets
- storage_buckets_create_credentials
- metrics_cpu_effective_total
- projects_networks_restricted_access
- storage_buckets_local
- loki
- acme
- internal_metrics
- cluster_join_token_expiry
- remote_token_expiry
- init_preseed
- storage_volumes_created_at
- cpu_hotplug
- projects_networks_zones
- network_txqueuelen
- cluster_member_state
- instances_placement_scriptlet
- storage_pool_source_wipe
- zfs_block_mode
- instance_generation_id
- disk_io_cache
- amd_sev
- storage_pool_loop_resize
- migration_vm_live
- ovn_nic_nesting
- oidc
- network_ovn_l3only
- ovn_nic_acceleration_vdpa
- cluster_healing
- instances_state_total
- auth_user
- security_csm
- instances_rebuild
- numa_cpu_placement
- custom_volume_iso
- network_allocations
- zfs_delegate
- storage_api_remote_volume_snapshot_copy
- operations_get_query_all_projects
- metadata_configuration
- syslog_socket
- event_lifecycle_name_and_project
- instances_nic_limits_priority
- disk_initial_volume_configuration
- operation_wait
- image_restriction_privileged
- cluster_internal_custom_volume_copy
- disk_io_bus
- storage_cephfs_create_missing
- instance_move_config
- ovn_ssl_config
- certificate_description
- disk_io_bus_virtio_blk
- loki_config_instance
- instance_create_start
- clustering_evacuation_stop_options
- boot_host_shutdown_action
- agent_config_drive
- network_state_ovn_lr
- image_template_permissions
- storage_bucket_backup
- storage_lvm_cluster
- shared_custom_block_volumes
- auth_tls_jwt
- oidc_claim
- device_usb_serial
- numa_cpu_balanced
- image_restriction_nesting
- network_integrations
- instance_memory_swap_bytes
- network_bridge_external_create
- network_zones_all_projects
- storage_zfs_vdev
- container_migration_stateful
- profiles_all_projects
- instances_scriptlet_get_instances
- instances_scriptlet_get_cluster_members
- instances_scriptlet_get_project
- network_acl_stateless
- instance_state_started_at
- networks_all_projects
- network_acls_all_projects
- storage_buckets_all_projects
- resources_load
- instance_access
- project_access
- projects_force_delete
- resources_cpu_flags
- disk_io_bus_cache_filesystem
- instance_oci
- clustering_groups_config
- instances_lxcfs_per_instance
- clustering_groups_vm_cpu_definition
- disk_volume_subpath
- projects_limits_disk_pool
- network_ovn_isolated
- qemu_raw_qmp
- network_load_balancer_health_check
- oidc_scopes
- network_integrations_peer_name
- qemu_scriptlet
- instance_auto_restart
- storage_lvm_metadatasize
- ovn_nic_promiscuous
- ovn_nic_ip_address_none
- instances_state_os_info
- network_load_balancer_state
- instance_nic_macvlan_mode
- storage_lvm_cluster_create
- network_ovn_external_interfaces
- instances_scriptlet_get_instances_count
- cluster_rebalance
- custom_volume_refresh_exclude_older_snapshots
- storage_initial_owner
- storage_live_migration
- instance_console_screenshot
- image_import_alias
- authorization_scriptlet
- console_force
- network_ovn_state_addresses
- network_bridge_acl_devices
- instance_debug_memory
- init_preseed_storage_volumes
- init_preseed_profile_project
- instance_nic_routed_host_address
- instance_smbios11
- api_filtering_extended
- acme_dns01
- security_iommu
- network_ipv4_dhcp_routes
- network_state_ovn_ls
- network_dns_nameservers
- acme_http01_port
- network_ovn_ipv4_dhcp_expiry
- instance_state_cpu_time
- network_io_bus
- disk_io_bus_usb
- storage_driver_linstor
- instance_oci_entrypoint
- network_address_set
- server_logging
- network_forward_snat
- memory_hotplug
- instance_nic_routed_host_tables
- instance_publish_split
- init_preseed_certificates
- custom_volume_sftp
- network_ovn_external_nic_address
- network_physical_gateway_hwaddr
api_status: stable
api_version: "1.0"
auth: trusted
public: false
auth_methods:
- tls
auth_user_name: anthony
auth_user_method: unix
environment:
  addresses: <redacted>
  architectures:
  - x86_64
  - i686
  certificate: <redacted>
  certificate_fingerprint: <redacted>
  driver: qemu | lxc
  driver_version: 9.0.4 | 6.0.4
  firewall: nftables
  kernel: Linux
  kernel_architecture: x86_64
  kernel_features:
    idmapped_mounts: "true"
    netnsid_getifaddrs: "true"
    seccomp_listener: "true"
    seccomp_listener_continue: "true"
    uevent_injection: "true"
    unpriv_binfmt: "false"
    unpriv_fscaps: "true"
  kernel_version: 6.1.0-37-amd64
  lxc_features:
    cgroup2: "true"
    core_scheduling: "true"
    devpts_fd: "true"
    idmapped_mounts_v2: "true"
    mount_injection_file: "true"
    network_gateway_device_route: "true"
    network_ipvlan: "true"
    network_l2proxy: "true"
    network_phys_macvlan_mtu: "true"
    network_veth_router: "true"
    pidfd: "true"
    seccomp_allow_deny_syntax: "true"
    seccomp_notify: "true"
    seccomp_proxy_send_notify_fd: "true"
  os_name: Debian GNU/Linux
  os_version: "12"
  project: default
  server: incus
  server_clustered: false
  server_event_mode: full-mesh
  server_name: <redacted>
  server_pid: 706
  server_version: "6.13"
  storage: btrfs
  storage_version: "6.2"
  storage_supported_drivers:
  - name: dir
    version: "1"
    remote: false
  - name: btrfs
    version: "6.2"
    remote: false
```

### Instance details

```
architecture: x86_64
config:
  environment.GPG_KEY: E3FF2839C048B25C084DEBE9B26995E310250568
  environment.HOME: /root
  environment.LANG: C.UTF-8
  environment.PATH: /usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
  environment.PYTHON_SHA256: 3126f59592c9b0d798584755f2bf7b081fa1ca35ce7a6fea980108d752a05bb1
  environment.PYTHON_VERSION: 3.9.21
  environment.TERM: xterm
  image.architecture: x86_64
  image.description: docker.io/santiagosayshey/profilarr (OCI)
  image.id: santiagosayshey/profilarr:latest
  image.type: oci
  oci.cwd: /app
  oci.entrypoint: gunicorn --bind 0.0.0.0:6868 app.main:create_app\(\)
  oci.gid: "0"
  oci.uid: "0"
  volatile.apply_template: create
  volatile.base_image: 16ea5790e612889e02e297094ae0b0d605c5770511a7564d6a43e663cafebf26
  volatile.cloud-init.instance-id: 7ee0f6b3-bead-4ed7-977b-3a0005bff7bb
  volatile.container.oci: "true"
  volatile.eth0.hwaddr: 10:66:6a:d2:ad:7b
  volatile.idmap.base: "0"
  volatile.idmap.next: '[{"Isuid":true,"Isgid":false,"Hostid":1000000,"Nsid":0,"Maprange":1000000000},{"Isuid":false,"Isgid":true,"Hostid":1000000,"Nsid":0,"Maprange":1000000000}]'
  volatile.last_state.idmap: '[]'
  volatile.uuid: 4a75f503-f17c-4c36-8841-b746125ae1da
  volatile.uuid.generation: 4a75f503-f17c-4c36-8841-b746125ae1da
devices: {}
ephemeral: false
profiles:
- default
stateful: false
description: ""
```

### Instance log

Strangely, on first execution of `incus info --show-log profilarr`, I get

> Error: Invalid PID '�'

And then if I run right after, I get many repeated lines of
> lxc profilarr 20250604014918.815 WARN     cgfsng - ../src/lxc/cgroups/cgfsng.c:__cgroup_tree_create:747 - File exists - Creating the final cgroup 10(lxc.monitor.profilarr) failed
lxc profilarr 20250604014918.815 WARN     cgfsng - ../src/lxc/cgroups/cgfsng.c:cgroup_tree_create:807 - File exists - Failed to create monitor cgroup 10(lxc.monitor.profilarr)

Finally followed by the last of line
> lxc profilarr 20250604014920.760 WARN     cgfsng - ../src/lxc/cgroups/cgfsng.c:cgfsng_monitor_destroy:925 - Device or resource busy - Failed to destroy 10(lxc.monitor.profilarr-25)

I doubt this is related, but figured I'd mention it.

### Current behavior

I'm getting an error when I launch this docker container [santiago/profilarr:latest](https://hub.docker.com/layers/santiagosayshey/profilarr/latest/images/sha256-c9b60fc4c69685d78d58ef39c600027323a99f1cf784db8ed164949f798ad9dd). It seems to be caused by the oci.entrypoint config value being escaped before passed through. I can work around it using the `raw.lxc` config mentioned in #1558.

> Failed to parse 'create_app\\(\\)' as an attribute name or function call.

```bash
$ incus start profilarr --console 
To detach from the console, press: <ctrl>+a q
[2025-06-03 12:59:54 +0000] [21] [INFO] Starting gunicorn 21.2.0
[2025-06-03 12:59:54 +0000] [21] [INFO] Listening at: http://0.0.0.0:6868 (21)
[2025-06-03 12:59:54 +0000] [21] [INFO] Using worker: sync
[2025-06-03 12:59:54 +0000] [22] [INFO] Booting worker with pid: 22
Failed to parse 'create_app\\(\\)' as an attribute name or function call.
[2025-06-03 12:59:54 +0000] [22] [INFO] Worker exiting (pid: 22)
[2025-06-03 12:59:54 +0000] [21] [ERROR] Worker (pid:22) exited with code 4
[2025-06-03 12:59:54 +0000] [21] [ERROR] Shutting down: Master
[2025-06-03 12:59:54 +0000] [21] [ERROR] Reason: App failed to load.
```

### Expected behavior

For OCI containers with parentheses to run without errors. At the least, I'd expect manually editing `oci.entrypoint` to  allow workaround.

### Steps to reproduce

1. Create the instance
```bash
$ incus init docker:santiagosayshey/profilarr:latest profilarr
```

2. Verify the value of `oci.entrypoint` already has the parens escaped
```bash
$ incus config get profilarr oci.entrypoint
gunicorn --bind 0.0.0.0:6868 app.main:create_app\(\)
```

3. Try to start the container and see the error
```bash
$ incus start profilarr --console 
To detach from the console, press: <ctrl>+a q
[2025-06-04 01:59:57 +0000] [21] [INFO] Starting gunicorn 21.2.0
[2025-06-04 01:59:57 +0000] [21] [INFO] Listening at: http://0.0.0.0:6868 (21)
[2025-06-04 01:59:57 +0000] [21] [INFO] Using worker: sync
[2025-06-04 01:59:57 +0000] [22] [INFO] Booting worker with pid: 22
Failed to parse 'create_app\\(\\)' as an attribute name or function call.
[2025-06-04 01:59:57 +0000] [22] [INFO] Worker exiting (pid: 22)
[2025-06-04 01:59:57 +0000] [21] [ERROR] Worker (pid:22) exited with code 4
[2025-06-04 01:59:57 +0000] [21] [ERROR] Shutting down: Master
[2025-06-04 01:59:57 +0000] [21] [ERROR] Reason: App failed to load.
```

4. I tried to manually fix through the editor or CLI, same result even though the config now does not show the escaped parens. So something must still be doing escaping along the way.
```bash
$ incus config set profilarr oci.entrypoint 'gunicorn --bind 0.0.0.0:6868 app.main:create_app()'

$ incus config get profilarr oci.entrypoint
gunicorn --bind 0.0.0.0:6868 app.main:create_app()

$ incus start profilarr --console 
To detach from the console, press: <ctrl>+a q
[2025-06-04 02:02:23 +0000] [21] [INFO] Starting gunicorn 21.2.0
[2025-06-04 02:02:23 +0000] [21] [INFO] Listening at: http://0.0.0.0:6868 (21)
[2025-06-04 02:02:23 +0000] [21] [INFO] Using worker: sync
[2025-06-04 02:02:23 +0000] [22] [INFO] Booting worker with pid: 22
Failed to parse 'create_app\\(\\)' as an attribute name or function call.
[2025-06-04 02:02:23 +0000] [22] [INFO] Worker exiting (pid: 22)
[2025-06-04 02:02:23 +0000] [21] [ERROR] Worker (pid:22) exited with code 4
[2025-06-04 02:02:23 +0000] [21] [ERROR] Shutting down: Master
[2025-06-04 02:02:23 +0000] [21] [ERROR] Reason: App failed to load.
```

5. Verify that raw.lxc works as expected
```bash
$ incus config set profilarr raw.lxc 'lxc.execute.cmd=gunicorn --bind 0.0.0.0:6868 app.main:create_app()'

$ incus config get profilarr raw.lxc
lxc.execute.cmd=gunicorn --bind 0.0.0.0:6868 app.main:create_app()

$ incus start profilarr --console 
To detach from the console, press: <ctrl>+a q
[2025-06-04 02:05:41 +0000] [21] [INFO] Starting gunicorn 21.2.0
[2025-06-04 02:05:41 +0000] [21] [INFO] Listening at: http://0.0.0.0:6868 (21)
[2025-06-04 02:05:41 +0000] [21] [INFO] Using worker: sync
[2025-06-04 02:05:41 +0000] [22] [INFO] Booting worker with pid: 22
2025-06-04 02:05:41 - app.init - INFO - Logging system initialized
2025-06-04 02:05:41 - app.init - INFO - Creating Flask application
```

--- a/internal/server/instance/drivers/driver_lxc.go
+++ b/internal/server/instance/drivers/driver_lxc.go
@@ -2367,15 +2367,22 @@ func (d *lxc) startCommon() (string, []func() error, error) {
 			}
 		}
 
+		// Compute the entrypoint string.
+		initCmd := shellquote.Join(entrypoint...)
+
+		// As we feed this to execve and not to a real shell, un-escape some sequences.
+		initCmd = strings.ReplaceAll(initCmd, "\\(", "(")
+		initCmd = strings.ReplaceAll(initCmd, "\\)", ")")
+
 		if len(entrypoint) > 0 && slices.Contains([]string{"/init", "/sbin/init", "/s6-init"}, entrypoint[0]) {
 			// For regular init systems, call them directly as PID1.
-			err = lxcSetConfigItem(cc, "lxc.init.cmd", shellquote.Join(entrypoint...))
+			err = lxcSetConfigItem(cc, "lxc.init.cmd", initCmd)
 			if err != nil {
 				return "", nil, err
 			}
 		} else {
 			// For anything else, run them under our own PID1.
-			err = lxcSetConfigItem(cc, "lxc.execute.cmd", shellquote.Join(entrypoint...))
+			err = lxcSetConfigItem(cc, "lxc.execute.cmd", initCmd)
 			if err != nil {
 				return "", nil, err
 			}

